cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterRebalanceAssemblyOperator.java

```diff
@@ -0,0 +1,804 @@
+                                                // TODO: There doesn't seem to be a way to retrieve the actual error message from the user tasks endpoint?
+                                                //       We may need to propose an upstream PR for this.
+                                                // TODO: Once we can get the error details we need to add an error field to the Rebalance Status to hold
+                                                //       details of any issues while rebalancing.
+                                                log.error("Rebalance ({}) optimization proposal has failed to complete", sessionId);
+                                                vertx.cancelTimer(t);
+                                                p.complete(new KafkaClusterRebalanceStatusBuilder()
+                                                        .withSessionId(sessionId)
+                                                        .addNewCondition().withType(State.NotReady.toString()).endCondition().build());
+                                                break;
+                                            case IN_EXECUTION: // Rebalance is still in progress
+                                                // We need to check that the status has been updated with the ongoing optimisation proposal
+                                                // The proposal field can be empty if a rebalance(dryrun=false) was called and the optimisation
+                                                // proposal was still being prepared (in progress). In that case the rebalance will start when
+                                                // the proposal is complete but the optimisation proposal summary will be missing.
+                                                if (freshClusterRebalance.getStatus().getOptimizationResult() == null ||
+                                                        freshClusterRebalance.getStatus().getOptimizationResult().isEmpty()) {
+                                                    log.info("Rebalance ({}) optimization proposal is now ready and has been added to the status", sessionId);
+                                                    // Cancel the timer so that the status is returned and updated.
+                                                    vertx.cancelTimer(t);
+                                                    p.complete(new KafkaClusterRebalanceStatusBuilder()
+                                                            .withSessionId(sessionId)
+                                                            .withOptimizationResult(taskStatusJson.getJsonObject("summary").getMap())
+                                                            .addNewCondition().withType(State.Rebalancing.toString()).endCondition().build());
+                                                }
+                                                ccAPIErrorCount.set(0);
```

Comment:
I did have one originally then thought it might be redundant. That count is for unsuccessful attempts to contact that CC server. If we have got here then we have successfully communicated and count can be reset.